Tracker/tracker.lua

```diff
@@ -3462,61 +3477,65 @@ function tracker:createNote( inChar )
         local k = row+1
         while( k < rows ) do
           if ( noteGrid[rows*chan+k] and not ( noteGrid[rows*chan+k] == noteToInterrupt ) ) then
-            break;
+            break
           end
           k = k + 1
-        end       
+        end
 
         -- Create the new note!
         self:addNote(row, k, chan, pitch, self.lastVel)
         self:playNote(chan, pitch, self.lastVel)
-      
+
         if ( noteGrid[rows*chan+k] ) then
-          if ( noteGrid[rows*chan+k] < -1 ) then    
+          if ( noteGrid[rows*chan+k] < -1 ) then
             self:deleteNote(chan, k)
           end
         end
-        
+
         -- If we interrupted a note, that note needs to be shortened / removed!
-        -- If we overwrote an OFF marker that was still here, then it needs to be removed as well. 
+        -- If we overwrote an OFF marker that was still here, then it needs to be removed as well.
         if ( noteToInterrupt ) then
           -- Note
           if ( noteToInterrupt > -1 ) then
             -- Shorten the note we are interrupting
             local pitch, vel, startppqpos, endppqpos = table.unpack( notes[noteToInterrupt] )
             endppqpos = self:rowToPpq(row)
-            
-            -- Check if we are at a legato point. Since we are interrupting a note, 
+
+            -- Check if we are at a legato point. Since we are interrupting a note,
             -- we have to maintain additional overlap
             if ( chan == 1 ) then
               local legato = self.legato
               if ( legato[row] and ( legato[row] > -1 ) ) then
-                endppqpos = endppqpos + tracker.magicOverlap          
+                endppqpos = endppqpos + tracker.magicOverlap
               end
             end
-            
+
             -- Set the new note length
             reaper.MIDI_SetNote(self.take, noteToInterrupt, nil, nil, nil, endppqpos, nil, nil, nil, true)
-          end          
+          end
         end
-        
+
         -- Were we overwriting an OFF marker?
         local idx = rows*chan+row
         if ( noteGrid[idx] and noteGrid[idx] < -1 ) then
           self:deleteNote(chan, row)
         end
       end
-      self.ypos = self.ypos + self.advance
+      if advance then

```

Comment:
As far as I can tell, what you wrote will work, but requires if statements in a fair number of places, with the risk that in a future addition, I or someone else could forget one. I agree that the way I previously solved it (setting advance to zero outside this function was worse than this), but maybe an alternative option would be to make the function:

function tracker:createNote(inChar, advance_override)

and at the start set:
local advance = advance_override or self.advance

and then just use the local advance in the rest of this function. Then advance can be set to zero if you wish to override it and omitted if not. What do you think?